[extension/jaegerremotesampling] Flaky test - TestStartAndShutdownRemote

Logs:

```
2022-04-06T19:52:18.7017926Z make -C ./extension/jaegerremotesampling test
2022-04-06T19:52:18.7035598Z make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/jaegerremotesampling'
2022-04-06T19:52:18.7089756Z go test -race -v -timeout 300s --tags="" ./...
2022-04-06T19:52:19.8199960Z === RUN   TestLoadConfig
2022-04-06T19:52:19.8201021Z --- PASS: TestLoadConfig (0.01s)
2022-04-06T19:52:19.8201622Z === RUN   TestValidate
2022-04-06T19:52:19.8202491Z === RUN   TestValidate/no_receiving_protocols
2022-04-06T19:52:19.8202884Z === RUN   TestValidate/no_sources
2022-04-06T19:52:19.8203215Z === RUN   TestValidate/too_many_sources
2022-04-06T19:52:19.8203777Z --- PASS: TestValidate (0.00s)
2022-04-06T19:52:19.8204379Z     --- PASS: TestValidate/no_receiving_protocols (0.00s)
2022-04-06T19:52:19.8204827Z     --- PASS: TestValidate/no_sources (0.00s)
2022-04-06T19:52:19.8205274Z     --- PASS: TestValidate/too_many_sources (0.00s)
2022-04-06T19:52:19.8205573Z === RUN   TestNewExtension
2022-04-06T19:52:19.8205908Z --- PASS: TestNewExtension (0.00s)
2022-04-06T19:52:19.8206458Z === RUN   TestStartAndShutdownLocalFile
2022-04-06T19:52:19.8206880Z --- PASS: TestStartAndShutdownLocalFile (0.00s)
2022-04-06T19:52:19.8207233Z === RUN   TestStartAndShutdownRemote
2022-04-06T19:52:19.8207674Z 2022/04/06 19:52:19 FATAL: [core] grpc: Server.RegisterService after Server.Serve for "jaeger.api_v2.SamplingManager"
2022-04-06T19:52:19.8208354Z FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling	0.035s
2022-04-06T19:52:19.8208813Z === RUN   TestMissingClientConfigManager
2022-04-06T19:52:19.8209241Z --- PASS: TestMissingClientConfigManager (0.00s)
2022-04-06T19:52:19.8209560Z === RUN   TestStartAndStop
2022-04-06T19:52:19.8209898Z --- PASS: TestStartAndStop (0.00s)
2022-04-06T19:52:19.8210192Z === RUN   TestEndpointsAreWired
2022-04-06T19:52:19.8210490Z === RUN   TestEndpointsAreWired/legacy
2022-04-06T19:52:19.8210798Z === RUN   TestEndpointsAreWired/new
2022-04-06T19:52:19.8211167Z --- PASS: TestEndpointsAreWired (0.00s)
2022-04-06T19:52:19.8211626Z     --- PASS: TestEndpointsAreWired/legacy (0.00s)
2022-04-06T19:52:19.8212094Z     --- PASS: TestEndpointsAreWired/new (0.00s)
2022-04-06T19:52:19.8212416Z === RUN   TestServiceNameIsRequired
2022-04-06T19:52:19.8212804Z --- PASS: TestServiceNameIsRequired (0.00s)
2022-04-06T19:52:19.8213149Z === RUN   TestErrorFromClientConfigManager
2022-04-06T19:52:19.8213583Z --- PASS: TestErrorFromClientConfigManager (0.00s)
2022-04-06T19:52:19.8214155Z PASS
2022-04-06T19:52:19.8214712Z ok  	github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal	(cached)
2022-04-06T19:52:19.8215086Z FAIL
2022-04-06T19:52:19.8328315Z make[2]: *** [../../Makefile.Common:46: test] Error 1
2022-04-06T19:52:19.8332324Z make[1]: *** [Makefile:162: for-all-target-./extension/jaegerremotesampling] Error 2
2022-04-06T19:52:19.8333144Z make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/jaegerremotesampling'
2022-04-06T19:52:19.8336794Z make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
2022-04-06T19:52:19.8337572Z make: *** [Makefile:72: gotest] Error 2
2022-04-06T19:52:19.8359336Z ##[error]Process completed with exit code 2.
```

--- a/extension/jaegerremotesampling/extension_test.go
+++ b/extension/jaegerremotesampling/extension_test.go
@@ -59,13 +59,14 @@ func TestStartAndShutdownRemote(t *testing.T) {
 
 	// create the mock server
 	server := grpc.NewServer()
+
+	// register the service
+	api_v2.RegisterSamplingManagerServer(server, &samplingServer{})
+
 	go func() {
 		err = server.Serve(lis)
 		require.NoError(t, err)
 	}()
-
-	// register the service
-	api_v2.RegisterSamplingManagerServer(server, &samplingServer{})
 
 	// create the config, pointing to the mock server
 	cfg := createDefaultConfig().(*Config)

